### PR TITLE
feat: configurable mining box

### PR DESCRIPTION
**Pull request by tomaszantas**
Description
---

Make MiningBox configurable:

* add Icons
* make it possible to override default statuses and styling

Add empty `WalletPasswordWizard` which looks like this:

image

... and will become this:

image

Motivation and Context
---

#12 
#13 

How Has This Been Tested?
---

https://user-images.githubusercontent.com/11715931/167091604-8124633b-f406-4cf5-93d2-6a8bc5767fc0.mov

### CODE REVIEW

**Review by corquaid — DISMISSED** [review]
LGTM!

File: applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/index.tsx [hunk 1]
```diff
@@ -151,10 +146,13 @@ const MiningBox = ({ node, children }: MiningBoxProps) => {
     },
   }
 
-  const currentState = {
-    ...defaultConfig,
-    ...defaultStates[nodeState.status],
-  }
+  const currentState = ObjectUtils.mergeDeep(
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/index.tsx]: this should at least be memoized to not mergeDeep every render

File: applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/styles.ts [hunk 2]
```diff
@@ -7,3 +7,19 @@ export const MiningBoxContent = styled.div`
   justify-content: space-between;
   flex: 1;
 `
+
+export const NodeIcons = styled.div<{ $color: string }>`
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/styles.ts:11]: what's the `$` for ? smells like angular

File: applications/launchpad/gui-react/src/utils/Object.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+const ObjectUtils = {
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/utils/Object.ts]: of all the code in here, this is what I would like to have tests for, it's too convoluted to leave without tests

**Review by tarnas14 — CHANGES_REQUESTED** [review]
I think this is overcomplicated and building a TariMiningContainer and MergedMiningContainer, which would manage state and render specific non-generic components would be more straightforward and easier to read and reuse

that said, enough effort has gone into it that I don't think we should scrap it, please add tests for mergeDeep or maybe use an npm package for it - production-tested, with tests and community support

File: applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/styles.ts (hunk 2 above)
Comment (tomaszantas) [reply applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/styles.ts:11]: Prop passed to the styled component with `$` prefix makes that it isn't added to the DOM.
(Already described in one of comments in previous PR)

SC complains particularly when we pass boolean. I found one issue in Tabs component, when we are passing `inverted: boolean`. This is what can be found in console or running `yarn test`:

image

This PR fixes this issue.

File: applications/launchpad/gui-react/src/utils/Object.ts (hunk 3 above)
Comment (tomaszantas) [reply applications/launchpad/gui-react/src/utils/Object.ts]: You are right! I've switched to the `deepmerge` package, so this file has been removed.

File: applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/index.tsx (hunk 1 above)
Comment (tomaszantas) [reply applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/index.tsx]: Good point! added `useMemo`

**Review by tarnas14 — APPROVED** [review]

**Pull request merged**